File: packages/style-analyzer/fixtures/naming-sample.ts
// Variables: camelCase
const userName = "Alice";
const accountBalance = 100;
let isActive = true;
let hasPermission = false;
const shouldRetry = true;

// Constants: SCREAMING_SNAKE
const MAX_RETRIES = 3;
const API_BASE_URL = "https://api.example.com";

// Functions: camelCase
function fetchUserProfile(userId: string) {
  return userId;
}

const calculateTotal = (items: number[]) => {
  return items.reduce((sum, item) => sum + item, 0);
};

// Types: PascalCase
interface UserProfile {
  name: string;
  age: number;
}

type ApiResponse = {
  data: unknown;
  status: number;
};

enum UserRole {
  Admin = "admin",
  Editor = "editor",
  Viewer = "viewer",
}

// Class: PascalCase
class HttpClient {
  private _baseUrl: string;

  constructor(baseUrl: string) {
    this._baseUrl = baseUrl;
  }

  async getData(endpoint: string) {
    return endpoint;
  }
}

// Parameters: camelCase
function processOrder(orderId: string, itemCount: number) {
  return { orderId, itemCount };
}

File: packages/style-analyzer/fixtures/naming-sample.py
# Variables: snake_case
user_name = "Alice"
account_balance = 100
is_active = True
has_permission = False

# Constants: SCREAMING_SNAKE
MAX_RETRIES = 3
API_BASE_URL = "https://api.example.com"

# Functions: snake_case
def fetch_user_profile(user_id: str):
    return user_id

def calculate_total(items: list[int]) -> int:
    return sum(items)

# Classes: PascalCase
class HttpClient:
    def __init__(self, base_url: str):
        self._base_url = base_url

    def get_data(self, endpoint: str):
        return endpoint

# Parameters: snake_case
def process_order(order_id: str, item_count: int):
    return {"order_id": order_id, "item_count": item_count}
